core/tools.js

```diff
@@ -2298,6 +2298,44 @@
 
 				appendParentFramePosition( frame.getWindow().getFrame() );
 			}
+		},
+
+		/**
+		 * Checks for conflicting plugins with the given one.
+		 *
+		 * If conflict occurs this function will send {@link CKEDITOR#warn console warning}
+		 * with `editor-plugin-conflict` error code. Order of a `conflicted` names is respected
+		 * where the last conflicted plugin has the highest priority and will be used in a warning
+		 * message.
+		 *
+		 * ```javascript
+		 * CKEDITOR.tools.detectPluginsConflict( editor, 'image', [ 'image2', 'easyimage' ] );
+		 * ```
+		 *
+		 * @param {CKEDITOR.editor} editor Editor searched for conflicting plugins.
+		 * @param {String} plugin Current plugin name.
+		 * @param {String[]} conflicted Names of plugins that are conflicted with a current plugin.
+		 * @return {Boolean} Returns true, if there is some conflict. Returns false otherwise.
+		 */
+		detectPluginsConflict: function( editor, plugin, conflicted ) {
+			var conflictPlugins = [];
+
+			for ( var i = 0; i < conflicted.length; i++ ) {
+				var pluginName = conflicted[ i ];
+
+				if ( editor.plugins[ pluginName ] ) {
+					conflictPlugins.push( pluginName );
```

Comment:
We don't need the array of all conflicting plugins. In fact we need only the last one. It could be simplified by reversing `conflicted` parameter (or reversing the loop) and returning as early as the first matched element.

Reply:
OTOH – IMO it's more intuitive to have the option with the highest priority at the beginning, not the end of the array.